/* BEGIN_COMMON_COPYRIGHT_HEADER
 * (c)LGPL2+
 *
 * LXQt - a lightweight, Qt based, desktop toolset
 * http://razor-qt.org
 *
 * Copyright: 2013 Christian Surlykke
 *            2014 Lu√≠s Pereira <luis.artur.pereira.gmail.com>
 *
 * This program or library is free software; you can redistribute it
 * and/or modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.

 * You should have received a copy of the GNU Lesser General
 * Public License along with this library; if not, write to the
 * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 * Boston, MA 02110-1301 USA
 *
 * END_COMMON_COPYRIGHT_HEADER */

#ifndef _MIMETYPEVIEWER_H
#define	_MIMETYPEVIEWER_H

#include <QDialog>
#include <QStringList>

#include <XdgMimeType>

#include "mimetypedata.h"
#include "ui_mimetypeviewer.h"

class QSettings;

namespace LxQt {
class SettingsCache;
}

class MimetypeViewer : public QDialog {
    Q_OBJECT
public:
    MimetypeViewer(QWidget *parent = 0);
    virtual ~MimetypeViewer();

private slots:
    void initializeMimetypeTreeView();
    void currentMimetypeChanged();
    void filter(const QString&);
    void chooseApplication();
    void dialogButtonBoxClicked(QAbstractButton *button);

private:
    void addSearchIcon();
    void loadAllMimeTypes();
    XdgMimeType m_CurrentMime;
    QSettings* mDefaultsList;
    LxQt::SettingsCache *mSettingsCache;
    Ui::mimetypeviewer widget;
    QStringList mediaTypes;
    QList <QTreeWidgetItem*> mItemList;
    QMap<QString, QTreeWidgetItem *> mGroupItems;
};

#endif	/* _MIMETYPEVIEWER_H */
